File: FlutterBookApp/lib/model/Book.dart
import 'package:flutter_book_app/model/Author.dart';

class Book {
  late int _id;
  late String _desc;
  late String _imgUrl;
  late String _isbn;
  late String _name;
  late double _point;
  late int _totalRead;
  late String _webUrl;
  late Author _author;

  Book(
      {required id,
      required desc,
      required imgUrl,
      required isbn,
      required name,
      required point,
      required totalRead,
      required webUrl,
      required author}) {
    _id = id;
    _desc = desc;
    _imgUrl = imgUrl;
    _isbn = isbn;
    _name = name;
    _point = point;
    _totalRead = totalRead;
    _webUrl = webUrl;
    _author = author;
  }

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
        id: json["id"] as String,
        desc: json["description"] as String,
        imgUrl: json["imgUrl"] as String,
        isbn: json["isbn"] as String,
        name: json["name"] as String,
        point: json["point"] as double,
        totalRead: json["totalRead"] as int,
        webUrl: json["webUrl"] as String,
        author: Author.fromJson(json));
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "description": desc,
      "imgUrl": imgUrl,
      "isbn": isbn,
      "name": name,
      "point": point,
      "totalRead": totalRead,
      "webUrl": webUrl,
      "author": _author.toJson()

    };
  }

  String get webUrl => _webUrl;

  set webUrl(String value) {
    _webUrl = value;
  }

  int get totalRead => _totalRead;

  set totalRead(int value) {
    _totalRead = value;
  }

  double get point => _point;

  set point(double value) {
    _point = value;
  }

  String get name => _name;

  set name(String value) {
    _name = value;
  }

  String get isbn => _isbn;

  set isbn(String value) {
    _isbn = value;
  }

  String get imgUrl => _imgUrl;

  set imgUrl(String value) {
    _imgUrl = value;
  }

  String get desc => _desc;

  set desc(String value) {
    _desc = value;
  }

  int get id => _id;

  Author get author => _author;

  set author(Author value) {
    _author = value;
  }

  @override
  String toString() {
    return 'Book{_id: $_id, _desc: $_desc, _imgUrl: $_imgUrl, _isbn: $_isbn, _name: $_name, _point: $_point, _totalRead: $_totalRead, _webUrl: $_webUrl}';
  }
}

File: neo4j-social-media/src/main/java/com/ahmeteminsaglik/neo4jsocialmedia/model/Book.java
package com.ahmeteminsaglik.neo4jsocialmedia.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.neo4j.core.schema.GeneratedValue;

import java.text.DecimalFormat;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Book {
    @Id
    @GeneratedValue
    private Long id;
    private String name;
    private double point;
    private String imgUrl;
    private int totalRead;
    private String description;
    private String isbn;
    private String webUrl;
    private Author author;
    private byte[] imageBytes;

    public byte[] getImageBytes() {
        return imageBytes;
    }

    public void setImageBytes(byte[] imageBytes) {
        this.imageBytes = imageBytes;
    }

    @Override
    public String toString() {
        return "Book{" +
                "id=" + id +
                ",\n author='" + author + '\'' +
                ",\n name='" + name + '\'' +
                ", point=" + point +
                ", imgUrl='" + imgUrl + '\'' +
                ", totalRead=" + totalRead +
                ", description='" + description + '\'' +
                ", isbn='" + isbn + '\'' +
                ", webUrl='" + webUrl + '\'' +
                '}';
    }

    public void setPoint(double point) {
        DecimalFormat decimalFormat = new DecimalFormat("#.#");
        decimalFormat.setMaximumFractionDigits(1);
        this.point = Double.parseDouble(decimalFormat.format(point));
    }

    public void setDescription(String description) {
        if (description != null) {
            String[] descArr = description.split("Contains");
            this.description = descArr[0];
        }
    }
}
